libzeth/mpc/groth16/powersoftau_utils.cpp

```diff
@@ -119,31 +119,28 @@ void write_powersoftau_fp2(
 
 } // namespace
 
-// Functions below are only implemented for the alt_bn128 curve type.
-using ppT = libff::alt_bn128_pp;
-
-void read_powersoftau_fr(std::istream &in, libff::Fr<ppT> &out)
+void read_powersoftau_fr(std::istream &in, libff::alt_bn128_Fr &out)
 {
     read_powersoftau_fp(in, out);
 }
 
-void read_powersoftau_g1(std::istream &in, libff::G1<ppT> &out)
+void read_powersoftau_g1(std::istream &in, libff::alt_bn128_G1 &out)
 {
     uint8_t marker;
     in.read((char *)&marker, 1);
 
     switch (marker) {
     case 0x00:
         // zero
-        out = libff::G1<ppT>::zero();
+        out = libff::G1<srs_pot_pp>::zero();
```

Comment:
`libff::alt_bn128_G1 &out` is used as type of the 2nd function parameter, so is it really needed to use `libff::G1<srs_pot_pp>` in the function? We know that for the function to be sound `srs_pot_pp` will need to be `libff::alt_bn128` here

Reply:
Same applies for some files below

Reply:
Used the `libff::alt_bn128_??` types in these fundamental serialization functions.